analyses/snv-callers/scripts/03-calculate_tmb.R

```diff
@@ -143,46 +141,109 @@ if (!dir.exists(opt$output)) {
 if (all(file.exists(c(tmb_coding_file, tmb_all_file)), !opt$overwrite)) {
   stop(paste0(tmb_coding_file, tmb_all_file, "both exist and --overwrite is not
               being used. Use --overwrite if you would like to overwrite these files."))
 }
-########################### Set up this consensus data ##########################
-# Print progress message
-message(paste("Reading in", opt$consensus, "MAF data..."))
 
-# Read in this MAF, skip the version number
-maf_df <- data.table::fread(opt$consensus, data.table = FALSE)
+######################## Obtain Mutect Strelka mutations #######################
+# Start up connection
+con <- DBI::dbConnect(RSQLite::SQLite(), opt$db_file)
+
+# Designate caller tables from SQL file
+strelka <- dplyr::tbl(con, "strelka")
+mutect <- dplyr::tbl(con, "mutect")
+
+# Specify the columns to join by
+join_cols <- c(
+  "Chromosome",
+  "Start_Position",
+  "Reference_Allele",
+  "Allele",
+  "Tumor_Sample_Barcode"
+)
+
+# Create the consensus for non-MNVs
+strelka_mutect_maf_df <- strelka %>%
+  dplyr::inner_join(mutect, by = join_cols)
+
+# Get Multi-nucleotide calls from mutect as SNVs
+split_mutect_df <- split_mnv(mutect)
+
+# join MNV calls with strelka
+strelka_mutect_mnv <- strelka %>%
+  dplyr::inner_join(split_mutect_df,
+    by = join_cols,
+    copy = TRUE
+  ) %>%
+  as.data.frame()
+
+if (opt$tcga) {
+  strelka_mutect_mnv <- strelka_mutect_mnv %>%
+    dplyr::mutate(Tumor_Sample_Barcode = substr(Tumor_Sample_Barcode, 0, 12))
+}
 
+# Split up MNVs
+strelka_mutect_maf_df <- strelka_mutect_maf_df %>%
+  dplyr::union_all(strelka_mutect_mnv,
+                   copy = TRUE
+  ) %>%
+  as.data.frame()
+
+########################### Set up metadata columns ############################
 # Print progress message
 message("Setting up metadata...")
 
-# Isolate metadata to only the samples that are in the datasets
-metadata <- readr::read_tsv(opt$metadata) %>%
-  dplyr::filter(Kids_First_Biospecimen_ID %in% maf_df$Tumor_Sample_Barcode) %>%
-  dplyr::distinct(Kids_First_Biospecimen_ID, .keep_all = TRUE) %>%
-  dplyr::rename(Tumor_Sample_Barcode = Kids_First_Biospecimen_ID)
+# Have to handle TCGA and PBTA metadata differently
+if (opt$tcga) {
+  # Format two fields of metadata for use with functions
+  metadata <- readr::read_tsv(opt$metadata) %>%
+    dplyr::mutate(
+      experimental_strategy = "WXS", # This field doesn't exist for this data, but all is WXS
+      short_histology = Primary_diagnosis
+    ) # This field is named differently
 
-# Make sure that we have metadata for all these samples.
-if (!all(unique(maf_df$Tumor_Sample_Barcode) %in% metadata$Tumor_Sample_Barcode)) {
-  stop("There are samples in this MAF file that are not in the metadata.")
+  # Manifest files only have first 12 letters of the barcode so we gotta chop the end off
+  strelka_mutect_maf_df <- strelka_mutect_maf_df %>%
+    dplyr::mutate(Tumor_Sample_Barcode = substr(Tumor_Sample_Barcode, 0, 12))
```

Comment:
Okay, you explain this here, but this is the second time you are doing it. Do you need it in the previous case, or is this one sufficient, as `strelka_mutect_mnv` is part of `strelka_mutect_maf_df`, if I understand line 201 correctly?

Reply:
This is actually unnecessary. The manifest doesn't need this done, only the MAF file does. I double checked that and now am deleting this step 👍